model/common/src/icon4py/model/common/utils/gt4py_field_allocation.py

```diff
@@ -5,20 +5,37 @@
 #
 # Please, refer to the LICENSE file in the root directory.
 # SPDX-License-Identifier: BSD-3-Clause
+from typing import Optional
+
 import gt4py.next as gtx
+from gt4py.next import backend
 
 from icon4py.model.common import type_alias as ta
 from icon4py.model.common.settings import xp
 
 
-def allocate_zero_field(*dims: gtx.Dimension, grid, is_halfdim=False, dtype=ta.wpfloat):
+def allocate_zero_field(
+    *dims: gtx.Dimension,
+    grid,
+    is_halfdim=False,
+    dtype=ta.wpfloat,
+    backend: Optional[backend.Backend] = None,
+):
     shapex = tuple(map(lambda x: grid.size[x], dims))
     if is_halfdim:
         assert len(shapex) == 2
         shapex = (shapex[0], shapex[1] + 1)
-    return gtx.as_field(dims, xp.zeros(shapex, dtype=dtype))
+    allocator = backend.allocator if backend else None
```

Comment:
I think contrary to what I said before, you don't need this you can pass the backend to the `allocator` kw argument directly. So
 ```
return gtx.as_field(dims, xp.zeros(shapex, dtype=dtype), allocator=backend)
```
Because the backend does have a `__gt_allocator__` property that maps to the `backend.allocator`